src/core_init_atmosphere/mpas_init_atm_cases.F

```diff
@@ -7140,4 +7147,63 @@ function read_text_array(dminfo, filename, xarray) result(ierr)
    end function read_text_array
 
 
+   !-----------------------------------------------------------------------
+   !  routine on_unit_sphere
+   !
+   !> \brief True if sphere surface area is close to unit sphere surface area (4*pii)
+   !> \author G. Dylan Dickerson
+   !> \date   03 Jan 2025
+   !> \details
+   !>
+   !>  This routine will use the global sum of cell areas to
+   !>  approximate the surface area of the sphere. If this surface area
+   !>  is within a factor of 4*pii, the mesh is likely still a unit
```

Comment:
The statement "within a factor of 4*pii" could be misinterpreted to mean that the surface area is +/- 4*pii of something.

Perhaps it would be better to reframe this description slightly to say that the routine returns true if it determines that the mesh is defined on a unit sphere -- which is the critical point -- and to then follow up with some notes that due to inexactness of arithmetic, the routine actually checks that the total sphere area is not larger than twice the area of a unit sphere.